narwhal/types/src/primary.rs

```diff
@@ -155,29 +157,74 @@ pub struct Header {
     pub signature: Signature,
 }
 
+impl Default for Header {
```

Comment:
is this a test-only impl?

Reply:
No, it's required by multiple objects that depend on the `Header`, e.g. `Certificate`.  This impl just signs them but I believe @joske implemented this so perhaps he had something else in mind.

Reply:
Although on closer inspection it seems that the wrapping objects only need their default impls for testing as well, this can be moved 👍

Reply:
yes, I needed it only for test code. Can't mark it as #[cfg(test)] as it's used across crates :-/

Reply:
fixed. Was only needed in one place, replaced with dedicated private method.